annotations/knative-annotations/src/main/java/io/dekorate/knative/generator/KnativeApplicationGenerator.java

```diff
@@ -75,6 +76,7 @@ default void on(ConfigurationSupplier<KnativeConfig> config) {
       Session session = getSession();
       session.configurators().add(config);
       session.handlers().add(new KnativeHandler(session.resources(), session.configurators()));
+      session.resources().groups().putIfAbsent(KNATIVE, new KubernetesListBuilder());
```

Comment:
Why?

I'd rather have generators not directly mess with the resources if possible.

Reply:
TBH this was ported from your [PR](https://github.com/dekorateio/dekorate/pull/407/files) and have no idea of what it's actually doing.
Same for: 
- https://github.com/dekorateio/dekorate/pull/426/files#diff-fb310640781383297c9efb65ecea11a6
- https://github.com/dekorateio/dekorate/pull/426/files#diff-e69a70d2513b994051c752bfba6f0125

Please, confirm if I should remove those too.

Reply:
Oooops! I really hate it when others do that to me!
Sorry, about that!

Reply:
keep them for now